components/dashboard/src/FromReferrer.tsx

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { Link } from "react-router-dom";
+
+export default function FromReferrer() {
+    const contextUrl = document.referrer;
+    if (contextUrl && contextUrl !== '') {
+        // Redirect to gitpod.io/#<contextUrl> to get the same experience as with direct call
+        const url = new URL(window.location.toString());
+        url.pathname = "/";
+        url.hash = contextUrl;
+        window.location.href = url.toString();
+        return <></>;
+    }
+    return <div className="lg:px-28 px-10 flex flex-col space-y-2">
+        <div className="px-6 py-3 flex justify-between space-x-2 text-gray-400 h-96">
+            <div className="flex flex-col items-center w-96 m-auto mt-40">
+                <h3 className="text-center pb-3 text-gray-500">No Referrer Found</h3>
+                <div className="text-center pb-6 text-gray-500">
+                    <p>It looks like you are trying to open a workspace, but the referrer URL is empty. This happens when the git hoster doesn't send the referrer header.
```

Comment:
**thought**: I still find the term _git hoster_ quite confusing. We could drop this sentence but this is fine, too.

![download](https://user-images.githubusercontent.com/120486/113406959-0fe59580-93b5-11eb-9680-8c792e297915.gif)